ci/docker/ubuntu_16.04/Dockerfile

```diff
@@ -0,0 +1,58 @@
+FROM ubuntu:16.04
+
+# Set environment variables - keep these alpha-sorted
+
+# Set up core apt requirements.
+# Do not add build requirements here.
+RUN apt-get -qq update && \
+    apt-get -qq install -y --no-install-recommends \
+    software-properties-common \
+    jq \
+    curl \
+    apt-transport-https
+
+# Add new repositories and their signing keys here.
+RUN add-apt-repository ppa:git-core/ppa \
+    && add-apt-repository "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-7 main" \
+    && curl -LSs "https://dl-ssl.google.com/linux/linux_signing_key.pub" | apt-key add - \
+    && curl -LSs "https://apt.llvm.org/llvm-snapshot.gpg.key" | apt-key add -
+
+# Install build dependency packages.
+# Keep these alpha-sorted.
+RUN apt-get -qq update && \
+    apt-get -qq install -y  --no-install-recommends \
+    # YACC needed for SWIG.
+    bison=2:3.0.4.dfsg-1 \
+    bash-completion=1:2.1-4.2ubuntu1.1 \
+    build-essential=12.1ubuntu2 \
+    ca-certificates=20170717~16.04.2 \
+    clang=1:3.8-33ubuntu3.1 \
```

Comment:
A lot of these dependencies are unnecessary for building Keanu. Something like https://github.com/improbable/db-r1-e2e-test/blob/master/ci/docker/ubuntu_16.04/Dockerfile would probably be sufficient.